### PR TITLE
[RHDX-131][RHDX-237] Add CSS and JS sourcemaps & remove Hugo from rhdp2

### PR DESCRIPTION
**Pull request by jordanpagewhite**
This adds CSS and JS sourcemaps to our rhdp2/rhd-frontend assets/build
process. This required removing some build logic that was moving dist
files around in Dockerfile.mp and dev/build-rhdp2-theme.sh, updating
paths across various FE build and config files and of course adding
to the existing node-sass command(s) in package.json to include the
building of sourcemaps.

### JIRA Issue Link
* https://projects.engineering.redhat.com/browse/RHDX-131
* https://projects.engineering.redhat.com/browse/RHDX-237

### Verification Process

* The build is successful
* Tests are green
* You can use CSS and JS sourcemaps in the PR environment and your local environment (if you build this branch locally)

You should see a directory structure like this in your Source tab:

Screen Shot 2020-02-04 at 10 03 37 PM

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
I need to push up another commit with changes to the rhd.settings.php file to disable Drupal's CSS and JS preprocessing on local environments. I am testing that on my local environment now and will push up a commit in a moment.

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
There was some change made in this PR that breaks the Log in (Keycloak only) link in the top-right corner of of the top navbar. @jordanpagewhite to investigate. To clarify, the actual authentication process, etc. works, but the e2e tests cannot successfully get to sso.stage.redhat.com via that link.

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by mindreeper2420** [comment]
I wasn't able to see the sourcemap using the demo URL. From what I could tell, it wasn't an issue with my browser (tried Firefox and Chrome) and tested against a site that I know distributes sourcemaps. Can anyone else verify the same?

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.libraries.yml [hunk 1]
```diff
@@ -25,11 +25,9 @@ base-theme:
       https://static.redhat.com/libs/redhat/redhat-theme/2/advanced-theme.css: { type: external }
       # NOTE: Provides layout classes for grids and alignments.
       css/pfe-layouts/pfe-layouts.min.css: { minified: true }
-      css/rhd.min.css: { minified: true }
-      css/rhd.microsites.css: { minified: true }
-      css/rhd.legacy.css: { minified: true }
-    theme:
-      css/rhdp2.css: {}
+      rhd-frontend/dist/css/rhd.min.css: { minified: true }
+      rhd-frontend/dist/css/rhd.microsites.css: { minified: true }
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.libraries.yml]: This file is empty, can we remove it here?
https://developer-preview-3486.ext.us-west.dc.preprod.paas.redhat.com/themes/custom/rhdp2/rhd-frontend/dist/css/rhd.microsites.css

**Comment by staceymosier** [comment]
> I wasn't able to see the sourcemap using the demo URL. From what I could tell, it wasn't an issue with my browser (tried Firefox and Chrome) and tested against a site that I know distributes sourcemaps. Can anyone else verify the same?

Discussed Drupal aggregation here; now that it's disabled in the PR site, you should be able to see maps.

File: _docker/drupal/managed-platform/ansible/templates/local/drupal/rhd.settings.php.j2 [hunk 2]
```diff
@@ -123,6 +123,10 @@ if($config['redhat_developers']['environment'] == 'local') {
     $settings['cache']['bins']['render'] = 'cache.backend.null';
     $settings['cache']['bins']['dynamic_page_cache'] = 'cache.backend.null';
     $settings['cache']['bins']['page'] = 'cache.backend.null';
+
```
Comment (staceymosier) [inline _docker/drupal/managed-platform/ansible/templates/local/drupal/rhd.settings.php.j2]: Add // phpcs:ignoreFile to the top of the rhd.settings.php file

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/managed-platform/ansible/templates/local/drupal/rhd.settings.php.j2 (hunk 2 above)
Comment (jordanpagewhite) [reply _docker/drupal/managed-platform/ansible/templates/local/drupal/rhd.settings.php.j2]: Added 👍

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.libraries.yml (hunk 1 above)
Comment (jordanpagewhite) [reply _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.libraries.yml]: Yes, removed 👍

**Comment by jordanpagewhite** [comment]
@staceymosier Could  you give this another review when you have a chance? I want to make sure all of your feedback was addressed properly.

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend/package.json [hunk 3]
```diff
@@ -8,26 +8,21 @@
...
-    "scripts:new": "uglifyjs-folder src/scripts/js/rhd -c -o dist/js/@rhd/rhd.min.js --source-map",
-    "scripts:old": "uglifyjs-folder src/scripts/js/2018 -c -o dist/js/@rhd/rhd.old.min.js --source-map",
-    "scripts:microsites": "uglifyjs-folder src/scripts/js/microsites -c -o dist/js/@rhd/rhd.microsites.min.js --source-map",
-    "scripts:ts": "tsc && copyfiles -u 4 \"dist/js/src/scripts/@rhd/**/*\" dist/js/node_modules/ && rollup -c rollup.config.js",
+    "scripts:dev": "npm run scripts:old && npm run scripts:new && npm run scripts:tsdev",
+    "scripts": "npm run scripts:old && npm run scripts:new && npm run scripts:ts && npm run scripts:copy",
+    "scripts:copy": "copyfiles \"node_modules/@patternfly/pfelement/pfelement.umd.min.js\" \"dist/js/patternfly/\" && copyfiles \"node_modules/@patternfly/pfelement/pfelement.umd.min.js.map\" \"dist/js/patternfly/\" && copyfiles \"node_modules/@patternfly/pfe-datetime/pfe-datetime.umd.min.js.map\" \"dist/js/patternfly/\" && copyfiles \"node_modules/@patternfly/pfe-datetime/pfe-datetime.umd.min.js.map\" \"dist/js/patternfly/\"",
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend/package.json]: There is a not found for /themes/custom/rhdp2/rhd-frontend/dist/js/patternfly/pfelement/pfelement.umd.min.js?q5cr95
which appears to cause a few js console errors on the homepage.

Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhd-frontend/package.json]: There is a not found for /themes/custom/rhdp2/rhd-frontend/dist/js/patternfly/pfelement/pfelement.umd.min.js?q5cr95
which appears to cause a few js console errors on the homepage.

**Comment by jordanpagewhite** [comment]
Screen Shot 2020-02-12 at 11 16 55 AM

I fixed the issue you pointed out (above) @staceymosier with the PFE dist filepaths. You should no longer see those console errors, and you should be able to see the src files in the Source tab of your Chrome inspector

**Pull request merged**